### DISCUSSION TITLE
Installation of Syft

### DISCUSSION THREAD

**Discussion by adilsheraz** [OpenMined/courses#373] [Course: Foundations of Private Computation]
While I run the command:
**`pip install syft`**
I face the following error:

ERROR: Command errored out with exit status 1:
   command: 'E:\ProgramData\envs\openmined-courses\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Dell E5540\\AppData\\Local\\Temp\\pip-install-pyj5oo4w\\netifaces_ffd79ab12b804ed389cdffccd68fe99c\\setup.py'"'"'; __file__='"'"'C:\\Users\\Dell E5540\\AppData\\Local\\Temp\\pip-install-pyj5oo4w\\netifaces_ffd79ab12b804ed389cdffccd68fe99c\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Dell E5540\AppData\Local\Temp\pip-wheel-lpastolv'
       cwd: C:\Users\Dell E5540\AppData\Local\Temp\pip-install-pyj5oo4w\netifaces_ffd79ab12b804ed389cdffccd68fe99c\
  Complete output (5 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'netifaces' extension
  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
  ----------------------------------------
  ERROR: Failed building wheel for netifaces
    ERROR: Command errored out with exit status 1:
     command: 'E:\ProgramData\envs\openmined-courses\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Dell E5540\\AppData\\Local\\Temp\\pip-install-pyj5oo4w\\netifaces_ffd79ab12b804ed389cdffccd68fe99c\\setup.py'"'"'; __file__='"'"'C:\\Users\\Dell E5540\\AppData\\Local\\Temp\\pip-install-pyj5oo4w\\netifaces_ffd79ab12b804ed389cdffccd68fe99c\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Dell E5540\AppData\Local\Temp\pip-record-p7sntvsl\install-record.txt' --single-version-externally-managed --compile --install-headers 'E:\ProgramData\envs\openmined-courses\Include\netifaces'
         cwd: C:\Users\Dell E5540\AppData\Local\Temp\pip-install-pyj5oo4w\netifaces_ffd79ab12b804ed389cdffccd68fe99c\
    Complete output (5 lines):
    running install
    running build
    running build_ext
    building 'netifaces' extension
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'E:\ProgramData\envs\openmined-courses\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Dell E5540\\AppData\\Local\\Temp\\pip-install-pyj5oo4w\\netifaces_ffd79ab12b804ed389cdffccd68fe99c\\setup.py'"'"'; __file__='"'"'C:\\Users\\Dell E5540\\AppData\\Local\\Temp\\pip-install-pyj5oo4w\\netifaces_ffd79ab12b804ed389cdffccd68fe99c\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Dell E5540\AppData\Local\Temp\pip-record-p7sntvsl\install-record.txt' --single-version-externally-managed --compile --install-headers 'E:\ProgramData\envs\openmined-courses\Include\netifaces' Check the logs for full command output.

**Comment by He4444** [comment]
You should download "Microsoft C++ Build Tools"
